packages/mermaid/src/utils.ts

```diff
@@ -940,3 +940,15 @@ export const getEdgeId = (
 ) => {
   return `${prefix ? `${prefix}_` : ''}${from}_${to}_${counter}${suffix ? `_${suffix}` : ''}`;
 };
+
+/**
+ * D3's `selection.attr` method doesn't officially support `undefined`.
+ *
+ * However, it seems if you do pass `undefined`, it seems to be treated as `null`
+ * (e.g. it removes the attribute).
+ */
+export function handleUndefinedAttr(
+  attrValue: Parameters<d3.Selection<BaseType, unknown, HTMLElement, any>['attr']>[1] | undefined
+) {
+  return attrValue ?? null;
+}
```

Comment:
I think we can move this to `packages/mermaid/src/rendering-util/rendering-elements/shapes/util.ts` since its only used in rendering-elements/

Reply:
Hmmmmm, it might be better to leave it here, since it would be used by a lot of the `diagrams/*/*rendering.js` files once we move them to `.ts`.

E.g. like https://github.com/mermaid-js/mermaid/blob/07de0907238db20abb50d905a71331d8394480cd/packages/mermaid/src/diagrams/gantt/ganttRenderer.js#L374

And I'm a bit reluctant to modify the import paths in ~60 files!